public/app/features/serviceaccounts/ServiceAccountsListPage.tsx

```diff
@@ -35,84 +36,168 @@ function mapStateToProps(state: StoreState) {
       fetchACOptions();
     }
   };
 
+  const onQueryChange = (value: string) => {
+    changeQuery(value);
+  };
+
+  const onStateFilterChange = (value: ServiceAccountStateFilter) => {
+    changeStateFilter(value);
+  };
+
+  const onRemoveButtonClick = (serviceAccount: ServiceAccountDTO) => {
+    setCurrentServiceAccount(serviceAccount);
+    setIsRemoveModalOpen(true);
+  };
+
+  const onServiceAccountRemove = async () => {
+    if (currentServiceAccount) {
+      deleteServiceAccount(currentServiceAccount.id);
+    }
+    onRemoveModalClose();
+  };
+
+  const onDisableButtonClick = (serviceAccount: ServiceAccountDTO) => {
+    setCurrentServiceAccount(serviceAccount);
+    setIsDisableModalOpen(true);
+  };
+
+  const onDisable = () => {
+    if (currentServiceAccount) {
+      updateServiceAccount({ ...currentServiceAccount, isDisabled: true });
+    }
+    onDisableModalClose();
+  };
+
+  const onEnable = (serviceAccount: ServiceAccountDTO) => {
+    updateServiceAccount({ ...serviceAccount, isDisabled: false });
+  };
+
+  const onTokenAdd = (serviceAccount: ServiceAccountDTO) => {
+    setCurrentServiceAccount(serviceAccount);
+    setIsAddModalOpen(true);
+  };
+
+  const onTokenCreate = async (token: ServiceAccountToken) => {
+    if (currentServiceAccount) {
+      createServiceAccountToken(currentServiceAccount.id, token, setNewToken);
+    }
+  };
+
+  const onAddModalClose = () => {
+    setIsAddModalOpen(false);
+    setCurrentServiceAccount(null);
+    setNewToken('');
+  };
+
+  const onRemoveModalClose = () => {
+    setIsRemoveModalOpen(false);
+    setCurrentServiceAccount(null);
+  };
+
+  const onDisableModalClose = () => {
+    setIsDisableModalOpen(false);
+    setCurrentServiceAccount(null);
+  };
+
   return (
     <Page navModel={navModel}>
       <Page.Contents>
-        <h2>Service accounts</h2>
-        <div className="page-action-bar" style={{ justifyContent: 'flex-end' }}>
+        <div className={styles.pageHeader}>
+          <h2>Service accounts</h2>
+          <div className={styles.apiKeyInfoLabel}>
+            <Tooltip
+              placement="bottom"
+              interactive
+              content={
+                <>
+                  API keys are now service Accounts with tokens. <a href="">Read more</a>
+                </>
+              }
+            >
+              <Icon name="question-circle" />
+            </Tooltip>
+            <span>Looking for API keys?</span>
+          </div>
+          {!noServiceAccountsCreated && contextSrv.hasPermission(AccessControlAction.ServiceAccountsCreate) && (
+            <LinkButton href="org/serviceaccounts/create" variant="primary">
+              Add service account
+            </LinkButton>
+          )}
+        </div>
+        <div className={styles.filterRow}>
           <FilterInput
-            placeholder="Search service account by name."
+            placeholder="Search service account by name"
             autoFocus={true}
```

Comment:
@Clarity-89 what do you think, should we auto focus search when navigating to the page? Looks strange a bit, it seems, we're not using this on other pages.

Reply:
Hmm, I think it's better to remove so it doesn't interfere with other accessibility features.